README-pt-br.md

```diff
@@ -0,0 +1,1730 @@
+
+De acordo com a especificação, o operador `typeof` retorna uma string de acordo com a [Table 35: `typeof` Operator Results](https://www.ecma-international.org/ecma-262/#table-35). Para `null`, objetos exóticos comuns e exóticos não padronizados, que não implementam `[[Call]]`, o retorno será a string `"object"`.
+
+Todavia, você poderá verificar o tipo de um objeto usando o método `toString`.
+
+```js
+Object.prototype.toString.call([]);
+// -> '[object Array]'
+
+Object.prototype.toString.call(new Date());
+// -> '[object Date]'
+
+Object.prototype.toString.call(null);
+// -> '[object Null]'
+```
+
+## Aumentando números magicamente
+
+```js
+999999999999999; // -> 999999999999999
+9999999999999999; // -> 10000000000000000
+
+10000000000000000; // -> 10000000000000000
+10000000000000000 + 1; // -> 10000000000000000
+10000000000000000 + 1.1; // -> 10000000000000002
+```
+
+### 💡 Explicação:
+
+Isso é causado pelo padrão IEEE 754-2008 para _Binary Floating-Point Arithmetic_ (Aritmética de binários de ponto flutuante). Nessa escala, ele arredonda para o número par mais próximo. Leia mais:
+
+- [**6.1.6** The Number Type](https://www.ecma-international.org/ecma-262/#sec-ecmascript-language-types-number-type)
+- [IEEE 754](https://en.wikipedia.org/wiki/IEEE_754) na Wikipedia
+
+## Precisão de  `0.1 + 0.2`
+
+Uma piada bastante conhecida. Uma adição de `0.1` e `0.2` é mortalmente precisa:
+
+```js
+0.1 +
+  0.2(
+    // -> 0.30000000000000004
+    0.1 + 0.2
+  ) ===
+  0.3; // -> false
+```
+
+### 💡 Explicação:
+
+A responsta para a pergunta [”Is floating point math broken?”](https://stackoverflow.com/questions/588004/is-floating-point-math-broken) no StackOverflow:
+
+> As constantes `0.2` and `0.3` no seu programa serão também aproximações dos seus valores verdadeiros. Isso ocorre quando o `double` mais próximo de `0.2` é maior que o número racional `0.2`, mas o `double` mais próximo de `0.3` é menor que o número racional `0.3`. A soma de `0.1` e `0.2` acaba sendo maior que o número racional `0.3`, e, portanto, discorda da constante em seu código.
+
+Esse problema é tão conhecido que existe um website chamado [0.30000000000000004.com](http://0.30000000000000004.com/). Isso ocorre em todas as linguagens que utilizam _floating-point math_ (matemática de ponto flutuante), não apenas no JavaScript.
+
+## Patching numbers
+
+Você pode adicionar seus próprios métodos em objetos como `Number` ou `String`.
+
+```js
+Number.prototype.isOne = function() {
+  return Number(this) === 1;
+};
+
+(1.0).isOne(); // -> true
+(1).isOne(); // -> true
+(2.0)
+  .isOne()(
+    // -> false
+    7
+  )
+  .isOne(); // -> false
+```
+
+### 💡 Explicação:
+
+Obviamente, você pode extender o objeto `Number` como qualquer outro no JavaScript. Contudo, não é recomendado se o comportamento do método definido não for parte da especificação. Aqui estão a lista de propriedades do `Number`:
```

Comment:
"Obviamente você pode extender o objeto `Number` como qualquer outro no JavaScript, contudo, não é recomendado se o comportamento do método definido não for parte da especificação"